README.md

```diff
@@ -2426,19 +2426,82 @@ ari.sounds.list()
 
 ### Events
 
-Event listeners can be registered on the client as well as on resource instances. Note that resource instance listeners are tied to the instance while client listeners receive all events of a given type regardless of which resource the event is for.
+Event listeners can be registered on the client as well as on resource instances.
+
+#### Client Events
+
+Client events are received for all events of a given type regardless of which resource the event is for.
+
+```javascript
+ari.on('StasisStart', function (event, channelInstance) {
+    // will be called for all channels that enter the Stasis application
+});
+```
+
+#### Resource Instance Events
+
+Resource instance events are only received for the given type and resource.
+
+Callbacks:
+
+```javascript
+var channel = ari.Channel();
+channel.on('StasisStart', function (event, channelInstance) {
+    // will only be called when the channel above enters the Stasis application
+});
+
+channel.originate(
+  {endpoint: 'PJSIP/endpoint', app: 'applicationName'},
+  function (err, channelInstance) {}
+);
+```
+
+Promises:
+
+```javascript
+var channel = ari.Channel();
+channel.on('StasisStart', function (event, channelInstance) {
+    // will only be called when the channel above enters the Stasis application
+});
+
+channel.originate({endpoint: 'PJSIP/endpoint', app: 'applicationName'})
+  .then(function (channelInstance) {})
+  .catch(function (err) {});
+```
+
+#### Managing Events
+
+When using events, it is important to remember that the callbacks you provide to handle events cannot be garbage collected until they are unregistered as shown below.
+
+```javascript
+var channel = ari.Channel();
+var handler = function (event, channel) {};
+
+channel.on('ChannelDtmfReceived', handler);
+
+// at some point in your application, remove the event listener to ensure the handler function can be garbage collected
+channel.removeListener('ChannelDtmfReceived', handler);
+// or remove all event listeners for a particular event type
+channel.removeAllListeners('ChannelDtmfReceived');
+```
+
+However, certain resource instance events are unregistered for you after a final event has been received for that resource to minimize possible memory leaks.
```

Comment:
Good idea. Currently only resources of type Channel, Bridge, Playback, and LiveRecording can be automatically cleaned up as they are the only resource that signal an end event.